src/components/common/FileUploadPdf.vue

```diff
@@ -0,0 +1,295 @@
+  readonly UPLOAD_FAILED_MESSAGE = 'An error occurred while uploading. Please try again.'
+
+  // Local properties
+  protected fileKey = null as string
+  protected errorMessages = [] as Array<string> // custom error messages
+  protected validFile = false
+
+  /**
+   * Clears input and resets validation errors.
+   * FUTURE: delete file from Minio
+   */
+  public reset (): void {
+    this.errorMessages = []
+    this.emitUpdateFile(null)
+    this.emitFileKey(null)
+  }
+
+  /** Resets input validation only. */
+  public resetValidation (): void {
+    this.errorMessages = []
+  }
+
+  /** Validates this component. */
+  public validate (): boolean {
+    // check that file exists if required
+    if (this.isRequired && !this.file) {
+      this.errorMessages = ['File is required']
+      return false
+    }
+    // return whether file is valid
+    return this.validFile
+  }
+
+  /** Called when user has changed the input file. */
+  protected async onChange (file: File): Promise<void> {
+    console.log('*** onChange(), file =', file)
+
+    // reset before validation
+    this.fileKey = null
+    this.errorMessages = []
+
+    // TODO:
+    // 1. validate file
+    // 2. upload file to Minio
+    // 3. emit file
+    // 4. emit file key
+
+    // *** TODO: need to check if required file exists???
+
+    if (file) {
+      this.errorMessages = ['Processing...']
+
+      // validate the file
+      console.log('*** validating file')
+      this.validFile = await this.validateFile(file)
+      if (!this.validFile) return
+
+      // upload the file
+      console.log('*** uploading file')
+      this.fileKey = await this.uploadFile(file)
+      if (!this.validFile) return
+
+      // update parent
+      this.emitUpdateFile(file)
+      this.emitFileKey(this.fileKey)
+    }
+  }
+
+  // Note: The validation is done this way as opposed to being all defined in the validation
+  //  rules above because Vuetify does not support async validation out of the box.  This was
+  //  needed to work around this limitation.  Vuelidate does support this but this would mean
+  //  introducing a component that is using a different validation approach or updating all
+  //  components to use Vuelidate.  Have decided to do this for the time being.
+  /**
+   * Validates the PDF file.
+   * @returns whether PDF file is valid
+   */
+  private async validateFile (file: File): Promise<boolean> {
+    // only perform PDF validation if there is a file
+    if (file) {
+      if (typeof file.arrayBuffer === 'undefined') return true
+
+      // verify file size
+      if (this.maxSize && (file?.size > (this.maxSize * 1024 * 1024))) {
+        this.errorMessages = [`Exceeds maximum ${this.maxSize} MB file size`]
+        return false
+      }
+
+      // try to retrieve file info
+      const fileInfo = await this.retrieveFileInfo(file).catch(() => null as PdfInfoIF)
+      if (fileInfo === null) {
+        this.errorMessages = ['Invalid PDF']
```

Comment:
New: check for valid PDF.

(Also on like 164 below, which can fail for the same reason.)